docs/modules/ROOT/pages/explanation/decisions/usernames.adoc

```diff
@@ -0,0 +1,57 @@
+= `User` object names in the OpenShift cluster
+
+Problem::
+We need to define an identifier which we use to name users which are synchronized from Keycloak to the OpenShift cluster.
++
+The identifier should have the following properties:
++
+* It should be stable for the lifetime of the user in Keycloak.
+* It must be suitable for use as an OpenShift resource name (namely for the OpenShift `User` resource)
+* It must be used to configure OpenShift `Group` membership when synchronizing groups from Keycloak.
```

Comment:
I would actually put those requirements into their own documents and refer to them via links.
`It should be stable for the lifetime of the user in Keycloak.` looks to be like requirement independent from any technology, so I guess it goes into the Usability category (this problem is actually independent of Keycloak itself).

The other two are depending/coupled to Keycloak and OpenShift, so I would propose to document those as functional requirements.
Right now, we don't have functional requirements documented yet, but it could be written as a Gherkin scenario that we already use in our Jira Stories. However, for the time being it's fine to leave them as is until we have a good template for functional requirements.

Reply:
I've added `docs/modules/ROOT/pages/references/quality-requirements/usability/user-arbitrary-name.adoc` (allowing users to choose arbitrary user names is clearly a usability requirement) and `docs/modules/ROOT/pages/references/quality-requirements/usability/user-name-stable.adoc`. However, I'm not sure from what perspective username stability can be considered a usability requirement.